app/components/files-list.js

```diff
@@ -0,0 +1,55 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { A } from '@ember/array';
+
+export default Component.extend({
+
+  /**
+   * Combine files with any associated actions so they can be displayed
+   */
+  filesWithActions: computed(
+    'files',
+    'submissionActions',
+    'submissionActions.@each.{key,description}',
+    function () {
+      const composition = [];
+
+      const actions = this.get('submissionActions') || A();
+
+      this.get('files').forEach((file) => {
```

Comment:
This is just a style point, but I prefer to `map` and return the mapped result directly rather than `forEach`'ing and pushing into an array.